src/pugixml.cpp

```diff
@@ -229,6 +231,8 @@ PUGI__NS_BEGIN
 
 	#ifdef PUGIXML_WCHAR_MODE
 		return wcscmp(src, dst) == 0;
+	#elif defined(PUGIXML_CHAR8_MODE)
+		return strcmp(reinterpret_cast<const char*>(src), reinterpret_cast<const char*>(dst)) == 0;
```

Comment:
Why is this not implemented in terms of `std::char_traits<pugi::char_t>::compare`?